lib/binding/kvdb_interface.c

```diff
@@ -445,14 +430,78 @@ hse_kvdb_close(struct hse_kvdb *handle)
     err = ikvdb_close((struct ikvdb *)handle);
     ev(err);
 
-    if (pidfile_remove(pfh) == -1 && !err)
-        err = merr(errno);
-
+    pidfile_remove(pfh);
     config_destroy(conf);
 
     return err;
 }
 
+hse_err_t
+hse_kvdb_attach(
+    const char *kvdb_home_tgt,
+    const char *kvdb_home_src,
+    const char *paths[HSE_MCLASS_COUNT])
+{
+    merr_t         err;
+    char           pidfile_path[PATH_MAX];
+    struct pidfh  *pfh_tgt = NULL, *pfh_src = NULL;
+    size_t         tlen, slen;
+
+    if (!kvdb_home_tgt || !kvdb_home_src) {
+        log_err("KVDB attach %s home must be provided", kvdb_home_src ? "target" : "source");
+        return merr(EINVAL);
+    }
+
+    tlen = strnlen(kvdb_home_tgt, PATH_MAX);
+    slen = strnlen(kvdb_home_src, PATH_MAX);
+    if (tlen == PATH_MAX || slen == PATH_MAX) {
+        log_err("KVDB attach %s home path length cannot be longer than PATH_MAX",
+                tlen == PATH_MAX ? "target" : "source");
+        return merr(ENAMETOOLONG);
+    }
+
+    if (tlen == 0 || slen == 0) {
+        log_err("KVDB attach %s home must be a non-zero length path",
+                tlen == 0 ? "target" : "source");
+        return merr(EINVAL);
```

Comment:
You could change these messages to refer to both target and source so you don't have to use `?:` to get the details right.  Something like: "KVDB attach target and source home dirs must both be non-zero length paths".

Reply:
Just wanted to be specific.. Sure, will fix.

Reply:
I didn't mean to indicate you should.... just saying IMO the generic form is fine. :)

Reply:
I like the generic form, it's cleaner!